File: insertion_sort/typescript/insertion_sort.ts
function insertion_sort(arr: number[], length: number) {
    for (let i = 1; i < length; i++) {
        const key = arr[i]
        let j = i - 1
        for (; j >= 0 && arr[j] > key; j--) {
            arr[j + 1] = arr[j]
        }
        arr[j + 1] = key
    }
}

export default insertion_sort

File: insertion_sort/go/insertion_sort.go
package insertionsort

import "cmp"

func InsertionSort[T cmp.Ordered](arr []T, n int) {
	for i := 1; i < n; i++ {
		key := arr[i]
		j := i - 1
		for ; j >= 0 && arr[j] > key; j-- {
			arr[j+1] = arr[j]
		}
		arr[j+1] = key
	}
}
